rskj-core/src/main/java/co/rsk/rpc/netty/ResponseSizeLimitContext.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2023 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.rpc.netty;
+
+import co.rsk.jsonrpc.JsonRpcError;
+import co.rsk.rpc.exception.JsonRpcResponseLimitError;
+import co.rsk.rpc.json.JsonResponseSizeLimiter;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.ethereum.rpc.exception.RskJsonRpcRequestException;
+
+public class ResponseSizeLimitContext implements AutoCloseable {
+
+    private static final ThreadLocal<ResponseSizeLimitContext> accumulatedResponseSize = new ThreadLocal<>();
+
+    private int size = 0;
+    private final int limit;
+
+    private ResponseSizeLimitContext(int limit) {
+        this.limit = limit;
+    }
+
+    private void add(int size) {
+
+        this.size += size;
+        if (this.size > limit) {
+            throw new JsonRpcResponseLimitError(limit);
+        }
+    }
+
+    private void add(JsonNode response) {
+        if (limit <= 0) {
+            return;
+        }
+
+        if (response != null) {
+            add(JsonResponseSizeLimiter.getSizeInBytesWithLimit(response, limit));
```

Comment:
it looks like that both `ResponseSizeLimitContext` and `JsonResponseSizeLimiter` accumulate response size. Can we do this only in one place? Eg. `ResponseSizeLimitContext` could do accumulation work and `JsonResponseSizeLimiter` (what about calling it something like `JsonResponseUtils`?) could simply return an approximated size?

Reply:
The `JsonResponseSizeLimiter` only stores the size during the calculation and then the result is being accumulated into the `ResponseSizeLimitContext`.